service/history/replication/executable_task.go

```diff
@@ -316,14 +332,37 @@ func (e *ExecutableTaskImpl) Resend(
 			tag.WorkflowRunID(retryErr.RunId),
 		)
 		// workflow is not found in source cluster, cleanup workflow in target cluster
-		return e.DeleteWorkflow(
+		return false, e.DeleteWorkflow(
 			ctx,
 			definition.NewWorkflowKey(
 				retryErr.NamespaceId,
 				retryErr.WorkflowId,
 				retryErr.RunId,
 			),
 		)
+	case *serviceerrors.RetryReplication:
+		// it is possible that resend will trigger another resend, e.g.
+		// 1. replicating a workflow which is a reset workflow (call this workflow `new workflow`)
+		// 2. base workflow (call this workflow `old workflow`) of reset workflow is deleted on
+		//	src cluster and never replicated to target cluster
+		// 3. when any of events of the new workflow arrive at target cluster
+		//  a. using base workflow info to resend until branching point between old & new workflow
+		//  b. attempting to use old workflow history events to replay for mutable state then apply new workflow events
+		//  c. attempt failed due to old workflow does not exist
+		//  d. return error to resend new workflow before the branching point
+
+		// handle 2nd resend error, then 1st resend error
```

Comment:
can you entail what this comment mean?

Reply:
src cluster:
T = 0 base workflow -> new workflow due to reset
T = 1 base workflow deleted due to retention
target cluster (brand new):
T = 2 receiving replication task for new workflow (e.g. migration)
T = 3 attempting to resend history to the point where base workflow & new workflow diverge, e.g. event ID 100
T = 4 attempting to use base workflow to replay for event 1-99, then apply 100, however base workflow does not exists on target cluster
T = 5 resend 1-99 using new workflow run ID from src cluster